crates/common/tedge_config_macros/impl/src/lib.rs

```diff
@@ -202,4 +202,18 @@ mod tests {
         })
         .unwrap();
     }
+
+    #[test]
+    fn error_message_suggests_fix_in_case_of_invalid_value() {
+        assert_eq!(generate_configuration(quote! {
+            http: {
+                #[tedge_config(default(value = Ipv4Addr::LOCALHOST))]
+                address: Ipv4Addr,
+            },
+        })
+                       .unwrap_err()
+                       .to_string(),
+                   "Unexpected expression, `default(value = ...)` expects a literal.\n\
+            Perhaps you want to use `#[tedge_config(default(variable = \"Ipv4Addr::LOCALHOST\"))]`?");
```

Comment:
Nitpicking.

To me an expression such as `Ipv4Addr::LOCALHOST` is a literal, i.e. an expression which evaluates to itself.
Maybe a lack of knowledge on Rust terminology.

That said the hint `Perhaps you want to use ... variable = \"Ipv4Addr::LOCALHOST\"` is wonderful!

Reply:
The reference explains quite nicely what a literal is: https://doc.rust-lang.org/beta/reference/expressions/literal-expr.html#:~:text=A%20literal%20expression%20is%20an,or%20some%20other%20evaluation%20rule.

> A literal expression is an expression consisting of a single token, rather than a sequence of tokens, that immediately and directly denotes the value it evaluates to, rather than referring to it by name or some other evaluation rule.

Reply:
Indeed, I missed this *single* token constraint.